File: SourceCode/Angular/src/app/core/domain-classes/sales-order-payment.ts
import { PurchaseOrder } from "./purchase-order";
import { SalesOrder } from "./sales-order";

export class SalesOrderPayment {
  id!: string;
  salesOrderId!: number
  salesOrder!: SalesOrder;
  paymentDate!: Date;
  referenceNumber!: string;
  orderNumber?: string;
  amount!: number;
  //PaymentMethod PaymentMethod { get; set; }
  note!: string;
  attachmentUrl!: string;
  attachmentData!: string;
  paymentMethod?: number;
  paymentType?: number;
}

File: SourceCode/SQLAPI/POS.Data/Entities/SalesOrder/SalesOrderPayment.cs
using POS.Data.Entities.Accounts;
using POS.Data.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace POS.Data
{
    public class SalesOrderPayment : BaseEntity
    {
        public Guid Id { get; set; }
        public Guid SalesOrderId { get; set; }
        [ForeignKey("SalesOrderId")]
        public SalesOrder SalesOrder { get; set; }
        public DateTime PaymentDate { get; set; }
        public string ReferenceNumber { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }
        public ACCPaymentMethod PaymentMethod { get; set; }
        public string Note { get; set; }
        public string AttachmentUrl { get; set; }
        public PaymentType PaymentType { get; set; }
    }
}
